src/ln/peer_handler.rs

```diff
@@ -1097,7 +1098,95 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 			}
 		};
 	}
-}
+
+	/// insure we recieved pong message from all peers or disconnect the peers then ping all peers
+	pub fn check_peer(&mut self)-> Result<(), PeerHandleError>{
+			Self::disconnect_if_no_pong(self)?;
+			Self::ping_peers(self)?;
+			Ok(())			
+	}	
+
+	// put a encoded ping message in all peers pending_outbound_buffer
+ 	fn ping_peers(&mut self) -> Result<(), PeerHandleError> {
+
+		for (Descriptor, Peer) in self.peers.lock().unwrap().peers.iter_mut(){
+			// create ping message
+			let ping = msgs::Ping {
+ 				ponglen: 64,
+ 				byteslen: 64
+  			};
+  			// create encoded ping message 
+  			let encoded_ping: &[u8] = &ping.encode();
+
+  			// put encoded_ping in the pending_outbound_buffer of each Peer
+			Peer.pending_outbound_buffer.push_back(Peer.channel_encryptor.encrypt_message(encoded_ping));
+	
+		}
+
+		Ok(())
+	}
+
+	//check Peers pending_read_buffer to see if anything in the vector resembles a pong message
+	// if there is no pong  message then we will disconnect the peer
+	fn disconnect_if_no_pong(&mut self) -> Result<(), PeerHandleError> {
```

Comment:
No need to add a function here, have a look on `DisconnectPeer` in `src/ln/msgs.rs`, you can look on `test_disconnect_peer` on how to use it.